specs/phase1/beacon-chain.md

```diff
@@ -614,106 +358,24 @@ def get_start_shard(state: BeaconState, slot: Slot) -> Shard:
     Return the start shard at ``slot``.
     """
     current_epoch_start_slot = compute_start_slot_at_epoch(get_current_epoch(state))
-    active_shard_count = get_active_shard_count(state)
-    if current_epoch_start_slot == slot:
-        return state.current_epoch_start_shard
-    elif slot > current_epoch_start_slot:
+    shard = state.current_epoch_start_shard
+    if slot > current_epoch_start_slot:
         # Current epoch or the next epoch lookahead
-        shard_delta = get_committee_count_delta(state, start_slot=current_epoch_start_slot, stop_slot=slot)
-        return Shard((state.current_epoch_start_shard + shard_delta) % active_shard_count)
-    else:
+        for _slot in range(current_epoch_start_slot, slot):
+            committee_count = get_committee_count_per_slot(state, compute_epoch_at_slot(Slot(_slot)))
+            active_shard_count = get_active_shard_count(state, compute_epoch_at_slot(Slot(_slot)))
+            shard = (shard + committee_count) % active_shard_count
+        return Shard(shard)
+    elif slot < current_epoch_start_slot:
         # Previous epoch
-        shard_delta = get_committee_count_delta(state, start_slot=slot, stop_slot=current_epoch_start_slot)
-        max_committees_per_slot = active_shard_count
-        max_committees_in_span = max_committees_per_slot * (current_epoch_start_slot - slot)
-        return Shard(
+        for _slot in list(range(slot, current_epoch_start_slot))[::-1]:
```

Comment:
```suggestion
        for _slot in list(range(slot - 1, current_epoch_start_slot))[::-1]:
```
Otherwise we will have the start shard at `slot + 1`.

Reply:
Also don't think this makes sense. Example: support `slot` is 95, `current_epoch_start_slot` is 100. `state.current_epoch_start_shard` represents the start shard pointer _at the start of_ slot 100. We want the start shard pointer _at the start of_ slot 95. So we want to process backwards, subtracting the shard counts for slots 99, 98, 97, 96, 95. The existing logic does this.

Reply:
You are correct. I misunderstood how Python ranges work.